File: JanperCab/5-JanperCab/ClientApp/src/app/_models/delivery-run-sheet/DeliveryRunSheetDto.ts
export class DeliveryRunSheetDto {
  static readonly BARCODE_PREFIX: string = 'RS';

  id: number;
  driverId: number;
  createdDate: Date;
  lockedDate: Date;
  deliveredDate: Date;

  get isEditable(): boolean {
    return !this.lockedDate && !this.deliveredDate;
  }

  getBarcodePrefix(): string {
    return DeliveryRunSheetDto.BARCODE_PREFIX;
  }
}

File: JanperCab/3-Application/Dtos/DeliveryRunSheet/DeliveryRunSheetDto.cs
using System;

namespace _3_Application.Dtos.DeliveryRunSheet
{
    public class DeliveryRunSheetDto
    {
        public int Id { get; set; }

        public int DriverId { get; set; }

        public int TruckId { get; set; }

        public DateTime CreatedDate { get; set; }

        public DateTime? LockedDate { get; set; }

        public DateTime? DeliveredDate { get; set; }
    }
}